core/domain/suggestion_registry.py

```diff
@@ -319,30 +318,124 @@ def accept(self, commit_message):
             self.final_reviewer_id, self.target_id, change_list,
             commit_message, is_suggestion=True)
 
-    @classmethod
-    def from_dict(cls, suggestion_dict):
-        """Return a Suggestion object of type SUGGESTION_TYPE_EDIT_STATE_CONTENT
-        from a dict.
 
-        Args:
-            suggestion_dict: dict. The dict representation of the suggestion.
+class SuggestionAddQuestion(BaseSuggestion):
+    """Domain object for a suggestion of type SUGGESTION_TYPE_ADD_QUESTION."""
 
-        Returns:
-            SuggestionEditStateContent. The corresponding Suggestion domain
-                object.
+    def __init__( # pylint: disable=super-init-not-called
+            self, suggestion_id, target_id, target_version_at_submission,
+            status, author_id, final_reviewer_id,
+            change_cmd, score_category, last_updated):
+        """Initializes an object of type SuggestionAddQuestion
+        corresponding to the SUGGESTION_TYPE_ADD_QUESTION choice.
+        """
+        self.suggestion_id = suggestion_id
+        self.suggestion_type = suggestion_models.SUGGESTION_TYPE_ADD_QUESTION
+        self.target_type = suggestion_models.TARGET_TYPE_TOPIC
+        self.target_id = target_id
+        self.target_version_at_submission = target_version_at_submission
+        self.status = status
+        self.author_id = author_id
+        self.final_reviewer_id = final_reviewer_id
+        self.change_cmd = question_domain.QuestionChange(change_cmd)
+        self.score_category = score_category
+        self.last_updated = last_updated
+
+    def validate(self):
+        """Validates a suggestion object of type SuggestionAddQuestion.
+
+        Raises:
+            ValidationError: One or more attributes of the SuggestionAddQuestion
+                object are invalid.
+        """
+        super(SuggestionAddQuestion, self).validate()
+
+        if self.get_score_type() != suggestion_models.SCORE_TYPE_QUESTION:
+            raise utils.ValidationError(
+                'Expected the first part of score_category to be %s '
```

Comment:
Probably want to have quotes around both "%s".

Reply:
Done